tests/lib/prepare.sh

```diff
@@ -256,32 +269,73 @@ prepare_classic() {
 }
 
 setup_reflash_magic() {
-        # install the stuff we need
-        distro_install_package kpartx busybox-static
-        distro_install_local_package "$GOHOME"/snapd_*.deb
-        distro_clean_package_cache
-
-        snap install "--${CORE_CHANNEL}" core
-
-        # install ubuntu-image
-        snap install --classic --edge ubuntu-image
-
-        # needs to be under /home because ubuntu-device-flash
-        # uses snap-confine and that will hide parts of the hostfs
-        IMAGE_HOME=/home/image
-        mkdir -p "$IMAGE_HOME"
-
+    # install the stuff we need
+    distro_install_package kpartx busybox-static
+    distro_install_local_package "$GOHOME"/snapd_*.deb
+    distro_clean_package_cache
+
+    # we cannot use "names.sh" here because no snaps are installed yet
+    core_name="core"
+    if is_core18_system; then
+        snap install "--${SNAPD_CHANNEL}" snapd
```

Comment:
Nitpick, also below, use `--channel=${SNAPD_CHANNEL}` explicitly to avoid `--help` or other weird things